### ISSUE TITLE
Release dependencies

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#188]
Release self-written or adapted repositories
- [x] `gtkmvc-dlr`
- [x] `yaml-configuration`
- [x] [`python-jsonconversion`](https://github.com/DLR-RM/python-jsonconversion)

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-10-17 12:14:13+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
For each of these, #187 must be fulfilled


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-11-08 14:55:52+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
For `gtkmvc-dlr`, the target is to merge our changes into the official `gtkmvc` repository. For this, we need to create pull requests for each feature/fix we implemented.


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-11-08 14:58:34+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Regarding `gtkmvc`:
1. Publish to GitHub
2. Write author
3. Ideally get the code merged


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-01-09 14:31:56+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
As Seth already created a rmc github account: https://github.com/DLR-RM we just have to wait for the permission of Freek.


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-02-03 07:28:51+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Before that, `python_yaml_configuration` needs to be prepared. E.g. a proper `setup.py` is needed and at least a `README`.

I'm currently doing so for `python-jsonconversion`.


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-02-03 07:51:29+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I created and tested a setup.py for the python_yaml_configuration package. It is able to create a source distribution (sdist) or a python egg (bdist_egg), which then can be installed via pip or easy_install. I also added a test and integrated it into buildbot and updated the README.

In the next meeting we should think about if we want a common PyPI account or separate ones.


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-02-16 13:09:30+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
So guys, I have forked the [gtkmvc3 repo](https://github.com/roboogle/gtkmvc3) of Roberto on our DLR/RM github.
In this repo you can find:
- the history on github is back to 2007
- it includes the version we have gtkmvc_dlr at data 2010-12-30 -> revision: [e7174c6](https://github.com/roboogle/gtkmvc3/tree/e7174c68a67a56e60d562095e9657e92d858b63b)
- last version of gtkmvc was 2013-09-17 (after this development started again 2015 because of our request) -> revision: [386af7c](https://github.com/roboogle/gtkmvc3/tree/386af7c)

I branched a [gtkmvc_dlr](https://rmc-github-server/Rbelder/gtkmvc3/tree/gtkmvc_dlr) branch locally (at revision: [e7174c6](https://rmc-github-server/Rbelder/gtkmvc3/tree/e7174c6)) and pushed it to my repo on our internal GitHub Enterprise. I moved our commits into the branch and skipped the rmpm stuff and maybe I can clean one or two commits but we can push this branch to the external Github if everybody is fine on Monday. I tested it and all unit-tests are fine with this. I have not found a option preserve the commits made on our own repo because we changed the folder structure.

Feel free to take a look into the gtkmvc_dlr repo and the branch gtkmvc_dlr.


Originally created by @Rbelder  at 2017-02-16 18:38:06+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Ok cool that we can work with the official gtkmvc3 repo. I just noticed that some of us already forked robertos gtkmvc3 repo in our public dlr-rm organization. We should probably only work with this one and delete our own repos (ricos and the common).


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-02-17 07:16:29+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Like mentioned before, there is a gtkmvc repo at my person git account on our internal github enterprise. The next days this repo will be moved to common/gtkmvc3. Tim and Martin I could not catch to do it but it will happen.

Now, locally on my site it looks like this and I would ask you to test it before I release it. 

origin  https://github.com/DLR-RM/gtkmvc3.git 
branches:
-   master
-   gtkmvc_dlr

rm_origin       https://rmc-github-server/Rbelder/gtkmvc3.git
branches:
-   master
-   gtkmvc_dlr
-   rmpm_gtkmvc_dlr

As already talked throw with @sebastian-brunner it is useful if some people test it. 

rmpm_gtkmvc-dlr is the rmpm release branch with pt-file and if needed other internal stuff. gtkmvc_dlr is the branch which is pushed to our public repo which is offically forked from roberto's [repo](https://github.com/roboogle/gtkmvc3).

So later you only have to change rm_origin to url=https://rmc-github-server/common/gtkmvc3.git

Added later:
The internal gtkmvc_dlr repo https://rmc-github-server/common/gtkmvc_dlr.git will be locked, and the releases of the package will continue on python-gtkmvc in the path [volume/software/common/foreign_packages](file:///internal_path/common/foreign_packages) from the always internal rmpm_gtkmvc_dlr branch.


Originally created by @Rbelder  at 2017-02-21 12:35:23+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Today @franzlst mentioned to me that a the name "gtkmvc_dlr" is not fitting and "dlr" would be enough because it is gtkmvc3 and so on. At first I thought maybe he is right and rename the branch is the best.

But after trying now to continue merging our stuff with the master of gtkmvc3 repo the reasoning about the name gtkmvc_dlr becomes quit intuitive.
The branch gtkmvc_dlr at the moment:
- has all changes we have done to gtkmvc of roberto
- and the master is merged into gtkmvc_dlr branch to the point were the gtk3 and python3 development of Roberto starts (so the last version of gtkmvc development of Roberto)

Therefore the naming is perfect. And if we will integrate all our changes in gtkmvc3 master fully that would result into a new branch named gtkmvc3_dlr. I will put some tags into the gtkmvc3 repo to find the respective versions easily. ;)

Added later:
Don't be weird by Roberto, because the hole history of gtkmvc and gtkmvc3 in one master branch the version in the files historically evolves as follows: 1.99.1, 1.99.2 (gtkmvc) and finally 1.0.0 (gtkmvc3). :smile: 


Originally created by @Rbelder  at 2017-02-21 15:16:37+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
- [x] As discussed in the last meeting we need a setup.py for gtkmvc -> we find out we do not need one


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-03-07 10:48:39+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Alternatively to releasing gtkmvc and thus having to create e.g. a `setup.py` for it, its manual explicitly mentions the option to ship the code with the application: http://pygtkmvc.sourceforge.net/docs/userman/install.html


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-03-09 06:54:06+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
So we just would have to adapt RAFCON's setup.py?


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-03-09 07:51:12+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I think so


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-03-09 08:35:45+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Number one :smile: 
https://github.com/DLR-RM/python-jsonconversion

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-03-31 10:59:43+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
python yaml conversion is also up on pypi, I am just waiting on stefan to create a repo on the public github

Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-04-13 15:12:25+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
So gtkmvc_dlr is now also on github with respective tar.gz-files for the releases gtkmvc_dlr_1.99.1 and gtkmvc_dlr_1.99.2. The issue is more or less finished. We can close it ceremonial when Sebastian has pushed to github.com, too. ;)

Originally created by @Rbelder  at 2017-04-13 15:37:25+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@franzlst Another setup.py for gtkmvc is not necessary right?

Originally created by @Rbelder  at 2017-04-13 15:39:04+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@Rbelder: I don't think so. You have tested it, right?
@sebastian-brunner: Any response from Stefan?

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-04-18 11:15:05+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@franzlst Sebastain has tested it on Thursday with the actual setup.py of RAFCON.

Originally created by @Rbelder  at 2017-04-18 11:44:34+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I think everybody of us should be able to verify this. Anyways, I can install RAFCON now with
```
python setup.py install --user
```
and also create a tar file with
```
python setup.py sdist
```


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-04-18 11:51:43+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
python yaml configuration is also now on the public github.

Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-05-01 13:41:35+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@Rbelder published gtkmvc-dlr on pypi

Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2017-05-02 12:02:45+00:00 (moved from RMC internal repository)
